### PR TITLE
[Tensor reorder][Quality][#issue 1476] Improve naming style and CTest design

### PR DESCRIPTION
**Pull request by aska-0096**
1. Improve naming style.
2. Modify CTest design, now only --all flag will conduct all the data type test while the default one is --float
3. Fix some small problems.

### CODE REVIEW

**Comment by atamazov** [comment]
[githib tip] @aska-0096 In the PR you can explicitly mention the issue it resolves. If you do so, then github will create a link _from the issue_ to the PR automatically. Like this:

This PR is intended to resolve issue #1476

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
```
Comment (atamazov) [inline .vscode/settings.json:1]: remove this file

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+build/
```
Comment (atamazov) [inline .gitignore:1]: ditto

**Review by atamazov — CHANGES_REQUESTED** [review]

File: test/tensor_reorder.cpp [hunk 3]
```diff
@@ -478,6 +507,7 @@ struct reorder_test : reorder_base
                                 order_3);
 
 #if MIOPEN_BACKEND_OPENCL
+            if(reorder_sol->IsSkippable()){
```
Comment (carlushuang) [inline test/tensor_reorder.cpp:510]: line:510 ~ 530 seem duplication of line:477~495?

File: test/tensor_reorder.cpp [hunk 4]
```diff
@@ -488,11 +518,16 @@ struct reorder_test : reorder_base
                                          nullptr,
                                          nullptr);
             EXPECT(status == CL_SUCCESS);
+            clReleaseMemObject(src_dev);
+            }
 #elif MIOPEN_BACKEND_HIP
+            if(reorder_sol->IsSkippable()){
             EXPECT(hipMemcpy(t_dst_gpu.data.data(),
-                             dst_dev,
+                             src_dev,
```
Comment (carlushuang) [inline test/tensor_reorder.cpp:526]: why copy `src_dev` to `t_dst_gpu`?

**Comment by carlushuang** [comment]
@aska-0096 please do clang-format when you submit code. Also, This branch `tensor_reorder` is already been used in https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1419 right? It's better you create a new branch e.g. `tensor_reorder_fix_issue_xxxx`, then create PR from the new branch. otherwise the commit history of this PR will have lots of duplications

File: test/tensor_reorder.cpp (hunk 3 above)
Comment (aska-0096) [reply test/tensor_reorder.cpp:510]: In this test,` dst_dev `memory workspace are controlled by` IsSkippable() `and `GetOutputTensorSize()`. When the reorder could be skipped, e.g. there are nothing changed in tensor memory layout after reorder operation, the workspace of `dst_dev` is assigned with `0 `memory space and we verify the result using the origin tensor on `src_dev`.

File: test/tensor_reorder.cpp (hunk 4 above)
Comment (aska-0096) [reply test/tensor_reorder.cpp:526]: when `IsSkippable() == true`, the reorder driver won't call reorder operation on GPU, and compare the origin tensor with CPU reordered tensor.

Comment (carlushuang) [reply test/tensor_reorder.cpp]: in `batched_transpose`, will still call the kernel even when `IsSkippable() == true`, if you actually call this kernel. The `IsSkippable()` is for the caller to optimize this process when `IsSkippable() == true`. In Other words, the gpu kernel should be OK to function well even when `IsSkippable() == true`. Is the design of generic transpose that gpu kernel will not compute correctly in such case?

Another thing is https://github.com/ROCmSoftwarePlatform/MIOpen/blob/c223c9461f198dddbd198b32f5a6b98a3c1dc27c/test/tensor_reorder.cpp#L521
`src_dev` will only be released here, when `IsSkippable() == true`, lead to memory leak.

File: test/tensor_reorder.cpp (hunk 3 above)
Comment (carlushuang) [reply test/tensor_reorder.cpp]: see blow

File: test/tensor_reorder.cpp (hunk 4 above)
Comment (aska-0096) [reply test/tensor_reorder.cpp]: I see. The ctest logic has been updated in new commits.

**Review by carlushuang — APPROVED** [review]
LGTM, please keep monitoring clang-format/tidy on CI

**Comment by aska-0096** [comment]
@atamazov Would to re-check that everything is ok with the tensor reorder feature? I'll keep monitoring on CI.

File: src/include/miopen/general_tensor_reorder_sol.hpp [hunk 5]
```diff
@@ -26,10 +26,10 @@
 #ifndef GUARD_GENERAL_MIOPEN_TENSOR_REORDER_SOL_HPP
 #define GUARD_GENERAL_MIOPEN_TENSOR_REORDER_SOL_HPP
 
-#include <miopen/miopen.h>
 #include <miopen/kernel_info.hpp>
 #include <miopen/op_kernel_args.hpp>
 #include <miopen/execution_context.hpp>
+#include <cstdlib>
```
Comment (shaojiewang) [inline src/include/miopen/general_tensor_reorder_sol.hpp:32]: I'm not pretty sure. It seems like that we need a ```<cstdint>``` here.

Comment (aska-0096) [reply src/include/miopen/general_tensor_reorder_sol.hpp]: Yes, I think `<cstdint>` is the proper header file here, changed it in the latest commit.

**Review by junliume — APPROVED** [review]

**Comment by aska-0096** [comment]
@atamazov ping for review.

**Review by shaojiewang — APPROVED** [review]
LGTM

**Review by carlushuang — APPROVED** [review]
LGTM

File: src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp [hunk 6]
```diff
@@ -23,7 +23,9 @@
  * SOFTWARE.
  *
  *******************************************************************************/
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
 #include <hip/hip_runtime.h>
+#endif
 #include <hip/hip_fp16.h>
```
Comment (junliume) [inline src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp:29]: :( I should not have missed this one!

**Review by atamazov — COMMENTED** [review]
Some fixes are required still. Thanks.

File: src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (junliume) [reply src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp:29]: done here: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1508/commits

File: src/kernels/gpu_general_tensor_reorder_kernel/order.hpp [hunk 7]
```diff
@@ -23,13 +23,14 @@
  * SOFTWARE.
  *
  *******************************************************************************/
+#include <cstdint>
```
[the remark this answers is not in the document]
Comment (aska-0096) [reply src/kernels/gpu_general_tensor_reorder_kernel/order.hpp:26]: Updated in #1515

File: src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (aska-0096) [reply src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp:29]: Thanks !

File: src/include/miopen/tensor_reorder_util.hpp [hunk 8]
```diff
@@ -245,40 +212,32 @@ TensorReorderSolutionConstructor(const ExecutionContext& ctx_,
...
             ctx_, data_type_, dim_0_, dim_1_, dim_2_, dim_3_);
     case 5:
-        return std::make_unique<WrapperBatchedTransposeSolution_3012>(
+        return std::make_unique<BatchedTransposeSolution_3012>(
             ctx_, data_type_, dim_0_, dim_1_, dim_2_, dim_3_);
-    default: return nullptr;
+    default: MIOPEN_THROW("Unsupported reorder sequence"); break;
     }
-    return nullptr;
+    return default_ptr;
```
[the remark this answers is not in the document]
Comment (aska-0096) [reply src/include/miopen/tensor_reorder_util.hpp:240]: Updated in #1515

File: src/include/miopen/tensor_reorder_util.hpp [hunk 9]
```diff
@@ -28,213 +28,180 @@
...
+                            uint32_t dim_3_,
+                            uint32_t order_0_,
+                            uint32_t order_1_,
+                            uint32_t order_2_,
+                            uint32_t order_3_)
 {
+    std::unique_ptr<TensorReorderAttributesBase> default_ptr;
+    if(!ctx_.use_hip_kernels)
+    {
+        return default_ptr;
```
[the remark this answers is not in the document]
Comment (aska-0096) [reply src/include/miopen/tensor_reorder_util.hpp:184]: updated in #1515

File: src/kernels/gpu_general_tensor_reorder_kernel/order.hpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (atamazov) [reply src/kernels/gpu_general_tensor_reorder_kernel/order.hpp:26]: 🟢 Resolved in #1515

File: src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (atamazov) [reply src/kernels/gpu_general_tensor_reorder_kernel/general_tensor_reorder.cpp:29]: 🟢 Resolved in #1508

File: src/include/miopen/tensor_reorder_util.hpp (hunk 9 above)
[the remark this answers is not in the document]
Comment (atamazov) [reply src/include/miopen/tensor_reorder_util.hpp:184]: 🟢 Resolved in #1515

File: src/include/miopen/tensor_reorder_util.hpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (atamazov) [reply src/include/miopen/tensor_reorder_util.hpp:240]: 🟢 Resolved in #1515

**Comment by atamazov** [comment]
@aska-0096 Thank you!

**Pull request merged**